pkg/neotest/basic.go

```diff
@@ -114,12 +114,20 @@ func (e *Executor) NewAccount(t *testing.T) Signer {
 }
 
 // DeployContract compiles and deploys contract to bc.
+// expectedID is expected contract ID for precalculated contract hash check
+// (can be set to negative value to omit check).
 // data is an optional argument to `_deploy`.
 // Returns hash of the deploy transaction.
-func (e *Executor) DeployContract(t *testing.T, c *Contract, data interface{}) util.Uint256 {
+func (e *Executor) DeployContract(t *testing.T, c *Contract, expectedID int32, data interface{}) util.Uint256 {
```

Comment:
Please don't introduce new parameters, you can get the ID from `Deploy` event generated by contract management contract.

Reply:
Ooopsie, looks like that's not the case. Still, you can retrieve contract data by hash.